drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -118,21 +118,34 @@ static void stm32_i2c_slave_event(const struct device *dev)
 	const struct i2c_stm32_config *cfg = dev->config;
 	struct i2c_stm32_data *data = dev->data;
 	I2C_TypeDef *i2c = cfg->i2c;
-	const struct i2c_slave_callbacks *slave_cb =
-		data->slave_cfg->callbacks;
+	const struct i2c_slave_callbacks *slave_cb;
+	struct i2c_slave_config *slave_cfg;
+	uint16_t slave_address;
+
+	/* Choose the right slave from the address match code */
+	slave_address = LL_I2C_GetAddressMatchCode(i2c) >> 1;
+	if (slave_address == data->slave_cfg->address) {
+		slave_cfg = data->slave_cfg;
+	} else if (slave_address == data->slave2_cfg->address) {
+		slave_cfg = data->slave2_cfg;
+	} else {
+		/* This should not happen, just return */
```

Comment:
maybe assert here?

Reply:
I have been looking at the code again, I think this could actually happen during a master transition in case a slave is attached. In that case just exiting is the correct thing to do I'll update the comment accordingly.

Reply:
Sorry, I am actually wrong, this code is only executed when `!data->master_active`, so this indeed should not apply. Is it fine adding an assert in interrupt code?

Reply:
I have found that `__ASSERT_NO_MSG` is used in other ISR code, so that should be fine. I have just pushed a new version using that.